api_core/google/api_core/protobuf_helpers.py

```diff
@@ -146,12 +158,40 @@ def get(msg_or_dict, key, default=_SENTINEL):
         raise KeyError(key)
 
     # If a subkey exists, call this method recursively against the answer.
-    if subkey and answer is not default:
+    if subkey is not None and answer is not default:
         return get(answer, subkey, default=default)
 
     return answer
 
 
+def _set_field_on_message(msg, key, value):
+    """Set helper for protobuf Messages."""
+    # Attempt to set the value on the types of objects we know how to deal
+    # with.
+    if isinstance(value, (collections.MutableSequence, tuple)):
+        # Clear the existing repeated protobuf message of any elements
+        # currently inside it.
+        while getattr(msg, key):
+            getattr(msg, key).pop()
+
+        # Write our new elements to the repeated field.
+        for item in value:
+            if isinstance(item, collections.Mapping):
+                getattr(msg, key).add(**item)
+            else:
+                getattr(msg, key).extend([item])
```

Comment:
Worth adding a note that `RepeatedCompositeContainer` doesn't support `append`?

Reply:
Done.